soc/arm/gigadevice/gd32a50x/soc.c

```diff
@@ -7,6 +7,20 @@
 #include <zephyr/init.h>
 #include <zephyr/irq.h>
 
+#define SRAM mmio_sram
+
```

Comment:
Two things to take in consideration:
1) Is it valuable check if `zephyr,sram` would be better option.
2) Is it possible to use memset ?

Reply:
> Is it valuable check if zephyr,sram would be better option.

I will check this one later

> Is it possible to use memset ?

I’ve tried this, and got NMI exception at startup.
```
void z_arm_platform_init(void)
{
	memset((void *)DT_REG_ADDR(DT_INST(0, SRAM)), 0, DT_REG_SIZE(DT_INST(0, SRAM)));
}
```
I think the reason is memset store data by char, but memory want written in 32bit.
```
void *memset(void *buf, int c, size_t n)
{
	unsigned char *d_byte = (unsigned char *)buf;
	unsigned char c_byte = (unsigned char)c;
              /* ...... */
	while (n > 0) {
		*(d_byte++) = c_byte;
		n--;
	}

	return buf;
}
```

![image](https://user-images.githubusercontent.com/10269596/207708284-328ee953-a24f-4428-8142-04d392a5f268.png)